unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -107,12 +107,27 @@ public static void freeMemory(long address) {
 
   public static void copyMemory(
     Object src, long srcOffset, Object dst, long dstOffset, long length) {
-    while (length > 0) {
-      long size = Math.min(length, UNSAFE_COPY_THRESHOLD);
-      _UNSAFE.copyMemory(src, srcOffset, dst, dstOffset, size);
-      length -= size;
-      srcOffset += size;
-      dstOffset += size;
+    // Check if dstOffset is before or after srcOffset to determine if we should copy
+    // forward or backwards. This is necessary in case src and dst overlap.
+    if (dstOffset < srcOffset) {
```

Comment:
I still have a question. Both `dstOffset` and `srcOffset` are offsets. They are not absolute addresses. 

Based on the code changes, I do not understand how it works. Could anybody give me a hint? Thank you!